### DISCUSSION TITLE
Setting Default Language for iPLAss GEM Module and AdminConsole

### DISCUSSION THREAD

**Discussion by khanh7516** [dentsusoken/iPLAss#1585] [Q&A]
I am not sure how to set the default language for the GEM Module and AdminConsole in iPLAss. In the Multilingual Settings section within the AdminConsole tenant, I have configured only two languages: Japanese and English. It is possible to switch between these two languages in the language settings of the GEM Module.

However, after logging into either the GEM Module or AdminConsole, the interface always defaults to English. I would like to know if there is a way to configure these applications so that the default language upon login is Japanese instead of English.

Thanks so much for your help!

**Comment by KoheiKGYM** [comment]
Hi @khanh7516 ,
thank you for using iPLAss!

The language setting of iPLAss is defined as follows (higher priority at the top).
1. "Language" property value of each user defined in “User entity”
2. Language setting stored in the session (when explicitly switched from GEM menu)
3. Browser language setting (Accept-Language header)
4. “Default Locale” setting in AdminConsole
    - https://iplass.org/docs/developerguide/multitenant/index.html#_多言語設定#:~:text=デフォルトロケール
5. Default setting of JVM  (value of user.language)

If you want to set the default language to Japanese, you can override all lower priority settings by specifying the "Language" property of the User entity to "ja”.  You need to login again after changing the property.
![Untitled](https://github.com/user-attachments/assets/5d2de380-e1a5-41cb-b0bb-bd5bb74dced2)
